mitxgraders/baseclasses.py

```diff
@@ -629,16 +646,47 @@ def check_response(self, answer, student_input, **kwargs):
                 graders when a grader is used as a subgrader in a ListGrader.
         """
 
+    inferring_answers = False
+
     def __call__(self, expect, student_input, **kwargs):
         """
         The same as AbstractGrader.__call__, except that we try to infer
         answers from expect argument if answers are not specified in the
-        grader configuration.
+        grader configuration. The actual inference is done by infer_answers,
+        which can be shadowed.
+        """
+        # If expect is provided, infer an answer if we either don't have an answer or
+        # are always inferring answers
+        if expect is not None and (self.inferring_answers or not self.config['answers']):
+            self.config['answers'] = self.infer_answers(expect)
```

Comment:
Two minor comments:

First: There's no reason to store this in `self.config['answers']` yet, right? The reason I ask is that I believe in all other parts of the library, `self.config` only ever stores validated data and `self.infer_answers(expect)` has not yet been validated. (Admittedly, it will be validated a few lines from now).

Second: At least in my mind, there is an internal distinction between an "answer" and an "expect value". `infer_answers` is actually inferring the expect value, and `schema_answers` is inferring the answer value from the expect value. Do you think it's worth making a naming change?

Reply:
Both reasonable. Addressed in commit.